### PR TITLE
feat: added rapidity getter in coffea/nanoevents/methods/vector.py

### PR DESCRIPTION
**Pull request by raeubaen**
Added a method to calculate rapidity for massive particles, for CMS higgs to gamma gamma columnar analysis (HiggsDNA) @emanueledimarco

### CODE REVIEW

File: coffea/nanoevents/methods/vector.py [hunk 1]
```diff
@@ -624,6 +624,12 @@ def pvec(self):
             behavior=self.behavior,
         )
 
+    @property
+    def rapidity(self):
+        pz = self.z
+        e = self.energy
+        return 0.5 * numpy.log((e + pz) / (e - pz))
```
Comment (lgray) [inline coffea/nanoevents/methods/vector.py]: `.5 * (numpy.log(e + pz)  - numpy.log(e - pz))` is more numerically stable

**Comment by lgray** [comment]
@raeubaen this will go into coffea 2023 - if you need it in coffea 0.7 for now please implement your own rapidity calculation!

**Comment by nsmith-** [comment]
Does this need to be backported to coffea 0.7?

**Comment by lgray** [comment]
Don't think so it's simple enough to implement as a free function or monkey patch.

**Pull request merged**